content/documentation/pages/6-recipes/2-rabbitmq/1-rabbit-source-sink.md

```diff
@@ -0,0 +1,119 @@
+
+Reading from and writing to RabbitMQ is a very common use-case. Primarily when the source and sink applications use spring Cloud Stream's RabbitMQ binder implementation, the configurations can be confusing.
+The goal of this recipe is to unpack the complexity step by step.
+
+Before we begin, let's describe the use-case requirements.
+
+_As a user, I'd like to,_
+
+- Consume `String` payload from a queue running in an external RabbitMQ cluster.
+- For every payload, I'd like to transform by converting the received `String` to uppercase.
+- Finally, I'd want to publish the transformed payload to a different queue, again running in the external RabbitMQ cluster.
+
+To further make it more interesting, we will also use the Spring Cloud Stream's RabbitMQ binder implementation in the source, processor, and sink applications.
+
+## Configuration
+
+There are two levels of RabbitMQ configurations required for this use-case.
+
+1. Configuration for RabbitMQ source and sink applications to connect to the external RabbitMQ cluster.
+2. Configuration of RabbitMQ binder properties at source, processor, and sink applications. We will use a locally running RabbitMQ at `127.0.0.1` (aka: `localhost`) for the binder.
+
+## Prerequisite
+
+1. Download the [`rabbit-source`](https://github.com/spring-cloud-stream-app-starters/rabbit/blob/master/spring-cloud-starter-stream-source-rabbit/README.adoc), [`transform-processor`](https://github.com/spring-cloud-stream-app-starters/transform/blob/master/spring-cloud-starter-stream-processor-transform/README.adoc), and [`rabbit-sink`](https://github.com/spring-cloud-stream-app-starters/rabbit/blob/master/spring-cloud-starter-stream-sink-rabbit/README.adoc) applications.
+
+   ```bash
+   wget https://repo.spring.io/release/org/springframework/cloud/stream/app/rabbit-source-rabbit/2.1.0.RELEASE/rabbit-source-rabbit-2.1.0.RELEASE.jar
+   ```
+
+   ```bash
+   wget https://repo.spring.io/release/org/springframework/cloud/stream/app/transform-processor-rabbit/2.1.0.RELEASE/transform-processor-rabbit-2.1.0.RELEASE.jar
+   ```
+
+   ```bash
+   wget https://repo.spring.io/release/org/springframework/cloud/stream/app/rabbit-sink-rabbit/2.1.0.RELEASE/rabbit-sink-rabbit-2.1.0.RELEASE.jar
+   ```
+
+2. Start RabbitMQ locally at `127.0.0.1`.
+3. Setup external RabbitMQ cluster and prepare the cluster connection credentials.
+
+## Deployment
+
+With all the prerequisites from the previous step complete, we can now start the three applications.
+
+### Source
+
+```bash
+java -jar rabbit-source-rabbit-2.1.0.RELEASE.jar --server.port=9001 --rabbit.queues=sabbyfooz --spring.rabbitmq.addresses=amqp://<USER>:<PASSWORD>@<HOST>:<PORT> --spring.rabbitmq.username=<USER> --spring.rabbitmq.password=<PASSWORD> --spring.cloud.stream.binders.rabbitBinder.type=rabbit --spring.cloud.stream.binders.rabbitBinder.environment.spring.rabbitmq.addresses=amqp://guest:guest@127.0.0.1:5672 --spring.cloud.stream.bindings.output.destination=rabzysrc
```

Comment:
Should we mention the properties that have prefix `spring.cloud.stream.binders` are the binder configuration properties and the `rabbitBinder` is just the configuration name?